Marlin/Marlin_main.cpp

```diff
@@ -1666,18 +1667,24 @@ static void clean_up_after_endstop_or_probe_move() {
 
     #else
 
-      feedrate = homing_feedrate[Z_AXIS];
-
-      current_position[Z_AXIS] = z;
-      line_to_current_position();
-      stepper.synchronize();
+      // Do the move at a high level.
+      if (current_position[Z_AXIS] < z) {
```

Comment:
I'm not sure about doing this here. The "blocking move" functions aren't necessarily restricted to leveling stuff, even though they have been up to now, so it should honor any requested Z move, and not just those that raise Z. If it turns out that all these functions are only used for probing, then perhaps we can keep this, but we should rename the functions to indicate that they manifest probe behavior.